### PR TITLE
[ticket/15579] Add ucp_front events

### PR DESCRIPTION
**Pull request by senky**
PHPBB3-15579

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15579

### CODE REVIEW

**Comment by vinny** [comment]
3.1 line is EOL.

**Comment by hanakin** [comment]
@senky pls do not change branches in Github interface it does not perform it properly and reopens randomly closed tickets in JIRA

**Comment by senky** [comment]
😅interesting. Alright, won't do again.

**Comment by senky** [comment]
!set Event

File: phpBB/includes/ucp/ucp_main.php [hunk 1]
```diff
@@ -77,6 +77,22 @@ function main($id, $mode)
...
 				if (!empty($forum_ary))
 				{
+					/**
+					 * Modify sql variables before query is processed
+					 *
+					 * @event core.ucp_main_front_modify_sql
+					 * @var string	sql_select	SQL select
+					 * @var string  sql_from	SQL from
+					 * @var array   forum_ary	Forum array
+					 * @since 3.1.13-RC1
```
Comment (vinny) [inline phpBB/includes/ucp/ucp_main.php]: wrong version

File: phpBB/includes/ucp/ucp_main.php [hunk 2]
```diff
@@ -171,8 +187,30 @@ function main($id, $mode)
...
+					/**
+					 * Add template variables to a front topics row.
+					 *
+					 * @event core.ucp_main_front_modify_template_vars
+					 * @var array	topicrow		Array containing the template variables for the row
+					 * @var array   row    	        Array containing the subscribed forum row data
+					 * @var int     forum_id        Forum ID
+					 * @var string  folder_img		Folder image
+					 * @var string  folder_alt      Alt text for the folder image
+					 * @since 3.1.13-RC1
```
Comment (vinny) [inline phpBB/includes/ucp/ucp_main.php]: same as above

File: phpBB/includes/ucp/ucp_main.php [hunk 3]
```diff
@@ -77,6 +77,22 @@ function main($id, $mode)
...
 				if (!empty($forum_ary))
 				{
+					/**
+					 * Modify sql variables before query is processed
+					 *
+					 * @event core.ucp_main_front_modify_sql
+					 * @var string	sql_select	SQL select
+					 * @var string  sql_from	SQL from
+					 * @var array   forum_ary	Forum array
+					 * @since 3.1.14-RC1
```
Comment (Nicofuma) [inline phpBB/includes/ucp/ucp_main.php]: 3.2.4

File: phpBB/includes/ucp/ucp_main.php [hunk 4]
```diff
@@ -171,8 +187,30 @@ function main($id, $mode)
...
+					/**
+					 * Add template variables to a front topics row.
+					 *
+					 * @event core.ucp_main_front_modify_template_vars
+					 * @var array	topicrow		Array containing the template variables for the row
+					 * @var array   row    	        Array containing the subscribed forum row data
+					 * @var int     forum_id        Forum ID
+					 * @var string  folder_img		Folder image
+					 * @var string  folder_alt      Alt text for the folder image
+					 * @since 3.1.14-RC1
```
Comment (Nicofuma) [inline phpBB/includes/ucp/ucp_main.php]: sorry I just commented the other, but 3.2.4 here too please :)

**Pull request merged**